mycroft/skills/media/__init__.py

```diff
@@ -0,0 +1,130 @@
+
+        intent = IntentBuilder('PrevIntent').require('PrevKeyword')
+        self.register_intent(intent, self.handle_prev)
+
+        intent = IntentBuilder('PauseIntent').require('PauseKeyword')
+        self.register_intent(intent, self.handle_pause)
+
+        intent = IntentBuilder('PlayIntent') \
+            .one_of('PlayKeyword', 'ResumeKeyword')
+        self.register_intent(intent, self.handle_play)
+
+        intent = IntentBuilder('CurrentlyPlayingIntent')\
+            .require('CurrentlyPlayingKeyword')
+        self.register_intent(intent, self.handle_currently_playing)
+
+    def _register_event_handlers(self):
+        """
+           Register event handlers for stopping currently playing media
+           when new media is started and handlers for lowering media volume
+           while mycroft is speaking.
+        """
+        self.emitter.on('mycroft.media.stop', self.handle_stop)
+        self.emitter.on('recognizer_loop:audio_output_start',
+                        self.lower_volume)
+        self.emitter.on('recognizer_loop:audio_output_end',
+                        self.restore_volume)
+
+    def handle_next(self, message):
+        """
+           handle_next() should be implemented by the skill to switch to next
+           song/channel/video in queue.
+        """
+        logger.debug('handle_next not implemented in ' + self.name)
+
+    def handle_prev(self, message):
+        """
+           handle_prev() should be implemented by the skill to switch to
+           previous song/channel/video in queue
+        """
+        logger.debug('handle_prev not implemented in ' + self.name)
+
+    def handle_currently_playing(self, message):
+        """
+           handle_currently_playing() should be implemented to tell the user
+           what is currently playing
+        """
+        logger.debug('handle_currently_playing not implemented in ' +
+                     self.name)
+
+    def play(self):
+        """ Stop currently playing media before starting the new. """
+        logger.info('Stopping currently playing media if any')
+        self.emitter.emit(Message('mycroft.media.stop'))
+
+    def handle_pause(self, message):
+        """ handle_pause() should pause currently playing media """
+        logger.debug('handle_pause not implemented in ' + self.name)
+
+    def handle_play(self, message):
+        """ handle_play() generic play handler. Should resume paused media
+            and/or implement generic playback functionality."""
+        logger.debug('handle_play not implemented in ' + self.name)
+
+    def handle_stop(self, message):
+        """
+           handle_stop() should be implemented to stop currently playing media
+        """
+        logger.debug('handle_stop not implemented in ' + self.name)
+
+    def stop(self):
+        self.handle_stop(None)
+
+    def lower_volume(self, message):
+        logger.debug('Lower volume not implemented in ' + self.name)
```

Comment:
We do have VolumeSkill. Let's integrate with it by sending a message to reduce the volume instead of reimplementing the same behaviour here. We should be able to do this by calling: 
`self.client.emit(Message("DecreaseVolumeIntent"))`